test/studies/ml/performance/mnist_cnn_no_batch.chpl

```diff
@@ -0,0 +1,165 @@
+import Chai as chai;
+import Tensor as tn;
+use Tensor;
+import Math;
+import MNIST;
+import Random;
+import IO;
+import Time;
+
+tn.seedRandom(0);
+
+config const dataPath = "ml-study/performance/data";
+
+// var net = new chai.Network(
+//     (
+//         new chai.Conv(1,20,3),
+//         new chai.MaxPool(),
+//         // new chai.SoftMax(13 * 13 * 8,10)
+//         new chai.Conv(20,10,3),
+//         new chai.MaxPool(),
+//         new chai.SoftMax(5 * 5 * 10,10)
+//     )
+// );
+
+// var net = new chai.Network(
+//     (
+//         new chai.Conv(1,6,5),   // 24
+//         new chai.MaxPool(),     // 12
+//         new chai.Conv(6,16,5),  // 8
+//         new chai.MaxPool(),     // 4
+//         new chai.SoftMax(4 * 4 * 16,10)
+//     )
+// );
+
+
+// This works
+// var net = new chai.Network(
+//     (
+//         new chai.Conv(1,8,3),
+//         new chai.MaxPool(),
+//         new chai.SoftMax(13 * 13 * 8,10)
+//     )
+// );
+
+// var net = new chai.Network(
+//     (
+//         new chai.Conv(1,8,3),
+//         new chai.MaxPool(),
+//         new chai.Conv(8,8,3),
+//         new chai.MaxPool(),
+//         new chai.SoftMax(5 * 5 * 8,10)
+//     )
+// );
+
+// // THIS IS MY BENCHMARK
+// var net = new chai.Network(
+//     (
+//         new chai.Conv(1,8,7),
+//         new chai.Conv(8,12,5),
+//         new chai.MaxPool(),
+//         new chai.SoftMax(10)
+//     )
+// );
+
+var net = new chai.Network(
+    (
+        new chai.Conv(1,8,4,stride=2),
+        new chai.Conv(8,12,5),
+        new chai.MaxPool(),
+        new chai.SoftMax(10)
+    )
+);
+
+proc forward(x: Tensor(?), lb: int) {
+    const output = net.forwardProp(x);
+    const loss = -Math.log(output[lb]);
+    const acc = tn.argmax(output.data) == lb;
+    return (output,loss,acc);
+}
+
+proc train(data: [] (Tensor(3),int), lr: real = 0.005) {
```

Comment:
I think it would be helpful to have all the training procs all in one file (probably MNISTTest) so they can be compared more directly. You could even have just one procedure that accepts a `batch: bool` argument.